api/v1/db/oracledb/terms-dao.js

```diff
@@ -0,0 +1,73 @@
+const appRoot = require('app-root-path');
+const _ = require('lodash');
+
+const { serializeTerms, serializeTerm } = require('../../serializers/terms-serializer');
+
+const { getConnection } = appRoot.require('api/v1/db/oracledb/connection');
+const { contrib } = appRoot.require('api/v1/db/oracledb/contrib/contrib');
+
+
+/**
+ * @summary Get current term code
+ * @function
+ * @param {Object} connection
+ * @returns {string} current term code
+ */
+const getCurrentTermCode = async (connection) => {
+  const { rows } = await connection.execute(contrib.getCurrentTerm());
+  if (_.isEmpty(rows) || rows.length > 1) {
+    return new Error('Expect a single object but got multiple results.');
+  }
+  return rows[0].termCode;
+};
+
+/**
+ * @summary Return a list of terms
+ * @function
+ * @param {string} query
+ * @returns {Promise} Promise object represents a list of terms
+ */
+const getTerms = query => new Promise(async (resolve, reject) => {
```

Comment:
I'm not completely sure on this, but since the anonymous `async` function here returns a Promise, wouldn't this be creating a Promise by wrapping another Promise? If so, should we simplify this to an async function or not? What are your thoughts on this?

Reply:
Nice, the way you suggested is pretty neat. Thanks!.